pkg/controller/deployment/deployment_controller.go

```diff
@@ -611,7 +627,13 @@ func (dc *DeploymentController) syncDeploymentStatus(allRSs []*extensions.Replic
 		return err
 	}
 	if d.Generation > d.Status.ObservedGeneration || d.Status.Replicas != totalActualReplicas || d.Status.UpdatedReplicas != updatedReplicas || d.Status.AvailableReplicas != availableReplicas {
-		return dc.updateDeploymentStatus(allRSs, newRS, d)
+		var cond *extensions.DeploymentCondition
+		if deploymentutil.IsProgressing(*d) {
+			// Mark the deployment as complete so that the lack of progress won't result in
+			// falsely marking it as failed.
+			cond = deploymentutil.NewCondition(extensions.DeploymentComplete, api.ConditionTrue, "Completed", "Deployment progress is complete")
+		}
+		return dc.updateDeploymentStatus(allRSs, newRS, d, cond)
```

Comment:
IIUC, if the update failed here (e.g. due to conflict) the deployment status condition will still be progressing and may be marked as timedout in the next round `syncDeployment` when we do the timedout check in L452.